src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -44,6 +44,12 @@ class FrameworkExtension extends Extension
      */
     public function load(array $configs, ContainerBuilder $container)
     {
+        // Check deprecation before the config is processed to ensure
+        // the setting has been explicitly defined in a configuration file.
+        if (isset($configs['csrf_protection']['field_name'])) {
```

Comment:
this will always be false, given that ``$configs`` is the array of configs coming from the different files.

If you want to do this before the processing, put it in the ``beforeNormalization`` of the node in the Configuration class